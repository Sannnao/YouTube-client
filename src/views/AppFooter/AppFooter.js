export default class AppFooter {
  static render() {
    const footer = document.createElement('footer');
    const firstItemContainer = document.createElement('div');
    const slider = document.createElement('div');

    const firstItem = document.createElement('div');
    const prevItem = document.createElement('div');
    const currentItem = document.createElement('div');
    const nextItem = document.createElement('div');

    const toolTip = document.createElement('div');

    footer.setAttribute('class', 'footer');
    firstItemContainer.setAttribute('class', 'first-item-container');
    slider.setAttribute('class', 'slider-container slider-hidden');

    firstItem.setAttribute('class', 'slider-item slider-item-interactive first-item slider-item-none');
    prevItem.setAttribute('class', 'slider-item prev-item slider-item-hidden');
    currentItem.setAttribute('class', 'slider-item current-item');
    nextItem.setAttribute('class', 'slider-item slider-item-interactive next-item');

    toolTip.setAttribute('class', 'tool-tip invisible-tool');

    firstItem.innerHTML = '<i class="fas fa-angle-double-left"></i>';
    prevItem.innerHTML = '<i class="fas fa-hand-point-left"></i>';
    nextItem.innerHTML = '<i class="fas fa-hand-point-right"></i>';
    currentItem.innerHTML = '1';
    toolTip.innerHTML = '1';

    firstItemContainer.appendChild(firstItem);
    slider.appendChild(toolTip);
    slider.appendChild(prevItem);
    slider.appendChild(currentItem);
    slider.appendChild(nextItem);

    footer.appendChild(firstItemContainer);
    footer.appendChild(slider);

    document.body.appendChild(footer);
  }
}
