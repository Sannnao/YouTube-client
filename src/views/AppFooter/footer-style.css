.footer {
  display: grid;
  grid-gap: 1%;
  grid-template-columns: 1fr 150px 1fr;
  justify-content: center;
  align-items: center;
}

.first-item-container {
  display: flex;
  flex-direction: row-reverse;
}

.slider-container {
  position: relative;
  grid-column: 2 / 3;
  display: flex;
  justify-content: space-around;
  align-items: center;
  width: 150px;
  height: 100%;
  box-shadow: 0 -1px 5px 0 rgb(190, 190, 190);
  border-radius: 10px;
}

.slider-hidden {
  display: none;
}

.slider-item {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 27px;
  height: 27px;
  font-weight: 700;
  font-size: 16px;
  background: linear-gradient(rgb(253, 222, 176),
                              rgb(253, 249, 185) 25%,
                              rgb(208, 230, 188) 50%,
                              rgb(164, 219, 193) 75%,
                              rgb(156, 222, 223));
  border-radius: 50%;
  border: 1px solid rgb(180, 180, 180);
  user-select: none;
  transition: box-shadow 0.3s, transform 0;
}

.slider-item-interactive:hover {
  font-size: 17px;
  box-shadow: 0 0 5px 0 rgb(117, 117, 117);
  transition: box-shadow 0.3s, transform 0;
  transform: scale(1.01, 1.01);
}

.slider-item-interactive:active {
  font-size: 16px;
  text-shadow: -1px -1px 0 rgb(110, 110, 110);
  box-shadow: inset 0 0 5px 0 rgb(110, 110, 110);
  transition: box-shadow 0.3s, transform 0;
  transform: scale(1.01, 1.01);
}

.slider-item-hidden {
  color: rgb(190, 190, 190);
  background: rgb(231, 231, 231);
  box-shadow: inset 0 0 5px 0 rgb(110, 110, 110);
}

.slider-item-none {
  display: none;
}

.tool-tip {
  position: absolute;
  z-index: 2;
  grid-column: 2 / 3;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 40px;
  height: 40px;
  font-size: 24px;
  background: rgb(104, 195, 152);
  border: 1px solid rgb(180, 180, 180);
  border-radius: 20px;
  box-shadow: 0 0 20px 0 rgb(163, 163, 163);
  transition: opacity cubic-bezier(1, -0.03, 1, .02) 2s, transform 1.5s;
  transform: translateY(0) scale(0.7);
  opacity: 0;
}

.tool-tip.show-tip {
  transition: opacity 0.1s;
  transform: translateY(-50px);
  opacity: 1;
}

.current-item {
  width: 30px;
  height: 30px;
  font-weight: 400;
  background: rgb(104, 195, 152);
  box-shadow: 0 0 5px 0 rgb(163, 163, 163);
}
