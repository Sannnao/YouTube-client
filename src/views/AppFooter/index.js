import AppFooter from './AppFooter';
import './footer-style.css';

export default AppFooter;
