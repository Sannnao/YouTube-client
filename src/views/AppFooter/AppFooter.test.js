import AppFooter from './AppFooter';

describe('AppFooter.render', () => {
  it('Should be an instance of Function', () => {
    expect(AppFooter.render).toBeInstanceOf(Function);
  });

  it('Should be render correctly', () => {
    AppFooter.render();
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
