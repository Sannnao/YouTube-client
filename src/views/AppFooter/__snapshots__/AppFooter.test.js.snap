// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter.render Should be render correctly 1`] = `"<footer class=\\"footer\\"><div class=\\"tool-tip invisible-tool\\">1</div><div class=\\"slider-container slider-hidden\\"><div class=\\"slider-item first-item slider-item-hidden\\">&lt;&lt;</div><div class=\\"slider-item prev-item slider-item-hidden\\">&lt;</div><div class=\\"slider-item current-item\\">1</div><div class=\\"slider-item next-item\\">&gt;</div></div></footer>"`;
